### PR TITLE
insert items from trash bags to autolathes

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
attacking an autolathe with a trash bag will now attempt to insert the items inside the trash bag to the autolathe, similar to trash bag to disposal bin.

Image of my trash bag full of "trash"
image

Gif of me inserting from the trash bag
https://gyazo.com/de36d9e85d5e00227de38f44a4b9108c

The resulting chat
image

## Why It's Good For The Game
Give an easier time to janitors to clean up and recycle!

## Changelog

:cl:
qol: you can now fill autolathes with items from trash bags
/:cl:

### CODE REVIEW

**Comment by the-orange-cow** [comment]
Would this also work with Rapid Part Exchange Devices and their bluespace variant?  Would be a nice way to harvest materials from lower tiered parts.

**Comment by jjpark-kb** [comment]
> Would this also work with Rapid Part Exchange Devices and their bluespace variant? Would be a nice way to harvest materials from lower tiered parts.

It could work with RPED/BSRPED, would look almost the same code-wise, but with the RPED/BSRPED instead of the trash bags.

**Comment by ArcaneDefence** [comment]
Is the recycler in disposals not good enough for this purpose?

**Comment by jjpark-kb** [comment]
> Is the recycler in disposals not good enough for this purpose?

this is quality of life, so if you choose to use the recycler, you may continue to do so.

File: code/game/machinery/autolathe.dm [hunk 1]
```diff
@@ -272,6 +272,13 @@
 		balloon_alert(user, "close the panel first!")
 		return FALSE
 
+	if(istype(attacking_item, /obj/item/storage/bag/trash))
+		for(var/obj/item/content_item in attacking_item.contents)
+			if(!do_after(user, 5, src))
```
Comment (Rohesie) [inline code/game/machinery/autolathe.dm:277]:
```suggestion
			if(!do_after(user, 0.5 SECONDS, src))
```

**Pull request merged**